### PR TITLE
Refactor beautifier.ts

### PR DESCRIPTION
**Pull request by szeck87**
This isn't ready yet, don't merge.  So far..

* LanguageManager class created and related methods moved to it from beautifier.ts
* OptionsManager class created and related methods moved to it from beautifier.ts
* New instances of the above classes are created when a new Unibeautify class is created, with existing languages and options variables passed into their respective constructors
* Existing public methods in beautifier.ts remain, but are marked as deprecated and just call the methods in the new classes
* LanguageQuery interface created for the findLanguages method
* Reduced code duplication within the findLanguage method

Closes https://github.com/Unibeautify/unibeautify/issues/74

### CODE REVIEW

File: src/LanguageManager.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import { unique, filterMultiCriteria } from "./utils";
+import { BeautifyData } from "./beautifier";
+import { Language } from "./language";
+
+export class LanguageManager {
+  constructor(protected _languages: Language[]) {}
```
Comment (Glavin001) [inline src/LanguageManager.ts]: Change `_languages` to `languages`.

File: src/LanguageManager.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      name: query.name,
+      namespace: query.namespace,
+      // tslint:disable-next-line:object-literal-sort-keys
+      extensions: query.extension,
+      atomGrammars: query.atomGrammar,
+      sublimeSyntaxes: query.sublimeSyntax,
+      vscodeLanguages: query.vscodeLanguage
+    };
+    const langs: Language[] = filterMultiCriteria(this._languages, filters);
+    return unique<Language>(langs);
```
Comment (Glavin001) [inline src/LanguageManager.ts]: I think `unique` should be within `filterMultiCriteria`

File: src/OptionsManager.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import {
+  Beautifier,
+  BeautifierOptionName,
+  BeautifierLanguageOptionComplex,
+  BeautifyOptionTransformSingleFunction,
+  BeautifyOptionTransform
+} from "./beautifier";
+
+export class OptionsManager {
+  constructor(protected _options: OptionsRegistry) {}
```
Comment (Glavin001) [inline src/OptionsManager.ts]: Change `_options` to `options`.

File: src/OptionsManager.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    return this;
+  }
+}
+
+export function optionKeys(
+  beautifier: Beautifier,
+  language: Language
+): BeautifierOptionName[] {
+  const beautifierOptions = beautifier.options[language.name];
+  // Transform options
```
Comment (Glavin001) [inline src/OptionsManager.ts]: Remove comment

File: src/beautifier.ts [hunk 5]
```diff
@@ -186,6 +186,14 @@ export class Unibeautify {
 
   */
   private beautifiers: Beautifier[] = [];
+  /**
+
+  */
+  private languageManager: LanguageManager = new LanguageManager(this.languages);
+  /**
+
+  */
```
Comment (Glavin001) [inline src/beautifier.ts]: No need for comments on a private variable: not public API and variable name clearly indicates usage.

Comment (szeck87) [reply src/beautifier.ts]: Are they needed for any of those?  I went along with the current https://github.com/Unibeautify/unibeautify/blob/master/src/beautifier.ts#L177-L188

File: src/LanguageManager.ts (hunk 2 above)
Comment (szeck87) [reply src/LanguageManager.ts]: I'm not sure how to do that.

File: src/beautifier.ts (hunk 5 above)
Comment (Glavin001) [reply src/beautifier.ts]: Yeah should have been removed in old code, too.

**Comment by szeck87** [comment]
Also, I'd like to make getLanguage permanently public, as I'd rather use that method in Atom Beautify instead of findLanguges.

File: src/LanguageManager.ts (hunk 2 above)
Comment (Glavin001) [reply src/LanguageManager.ts]: What I mean is:

# Before

```
const langs: Language[] = filterMultiCriteria(this.languages, filters);
return unique<Language>(langs);
```

# After

```
return filterMultiCriteria(this.languages, filters);

// ... inside filterMultiCriteria method
function filterMultiCriteria(langs, filters) {
  return unique<Language>(previousReturnValue);
}
```

Comment (szeck87) [reply src/LanguageManager.ts]: I'm trying to keep filterMultiCriteria generic and reusable, putting unique in there with language would break that.

**Comment by Glavin001** [comment]
@szeck87 : I think exposing more helper methods is a good idea. 👍

**Comment by szeck87** [comment]
@Glavin001 I included a fix to #85 here.  Anything outstanding here?

**Comment by Glavin001** [comment]
Awesome work, @szeck87 ! Thanks again!! 🎉

**Pull request merged**